docker/Dockerfile

```diff
@@ -1,80 +1,106 @@
-RUN cd zstd-1.1.0 && make install PREFIX='/usr'
-RUN cd /usr/src/gtest && cmake . && make && mv libgtest* /usr/lib/
+RUN cd / && wget https://github.com/facebook/zstd/archive/v1.1.0.tar.gz && \
+    tar xf v1.1.0.tar.gz && cd zstd-1.1.0 && \
+    make install PREFIX='/usr'
 
 # TileDB
-RUN cd / && wget https://github.com/TileDB-Inc/TileDB/archive/0.6.1.tar.gz && tar xf 0.6.1.tar.gz
-RUN cd TileDB-0.6.1 && mkdir build && cd build && cmake .. && make -j $(cat /proc/cpuinfo | wc -l) && make install
-RUN rm -rf /TileDB-0.6.1
+RUN cd / && wget https://github.com/TileDB-Inc/TileDB/archive/1.3.1.tar.gz && \
+    tar xf 1.3.1.tar.gz && cd TileDB-1.3.1 && mkdir build && cd build && \
+    ../bootstrap --prefix=/usr/local/ && make -j && make install-tiledb
+RUN rm -rf /TileDB-1.3.1
 
-# Valijson
-RUN cd / && git clone https://github.com/tristanpenman/valijson.git
-RUN cd valijson && cp -r include/* /usr/local/include/
+# Faiss
+RUN cd / && \
+    wget https://github.com/facebookresearch/faiss/archive/v1.4.0.tar.gz && \
+    tar xf v1.4.0.tar.gz && \
+    cd faiss-1.4.0 && rm CMakeLists.txt && \
+    wget https://gist.githubusercontent.com/luisremis/758c71c6a86cb3e4fe2311c415e07547/raw/3c51b5ed7d3f16a4d684d9ea524602c9f1f8593a/CMakeLists.txt && \
+    mkdir build && cd build && cmake ../ && make -j
+RUN mkdir /usr/local/include/faiss/
+RUN cp -r faiss-1.4.0/* /usr/local/include/faiss/
+RUN mv faiss-1.4.0/build/lib/libfaiss.so /usr/local/lib/
+RUN rm -rf faiss-1.4*
 
-# OpenCV
-RUN mkdir /opencv && cd /opencv && wget -O opencv.zip https://github.com/opencv/opencv/archive/3.3.0.zip
-RUN cd /opencv && unzip opencv.zip && cd opencv-3.3.0 && mkdir build && cd build && cmake -D CMAKE_BUILD_TYPE=Release -D CMAKE_INSTALL_PREFIX=/usr/local .. && make -j $(cat /proc/cpuinfo | wc -l) && make install && rm -rf /opencv
+# PMGD install
+RUN cd / && wget https://github.com/intellabs/pmgd/archive/v2.0.0.tar.gz && \
+    tar xf v2.0.0.tar.gz && mv pmgd-2.0.0 pmgd && cd pmgd && \
+    make PMOPT=MSYNC
 
 # VCL install
-RUN cd / && wget https://github.com/intellabs/vcl/archive/v0.1.0.tar.gz && tar xf v0.1.0.tar.gz
-RUN mv vcl-0.1.0 vcl && cd vcl && scons -j16
+RUN cd / && wget https://github.com/intellabs/vcl/archive/v1.0.0.tar.gz && \
+    tar xf v1.0.0.tar.gz && mv vcl-1.0.0 vcl && cd vcl && scons -j16
 
-# PMGD install
-RUN cd / && wget https://github.com/intellabs/pmgd/archive/v1.0.0.tar.gz && tar xf v1.0.0.tar.gz
-RUN mv pmgd-1.0.0 pmgd && cd pmgd && make
-RUN rm -rf /*.tar.gz
+# Valijson
+RUN cd / && git clone https://github.com/tristanpenman/valijson.git && \
+    cd valijson && cp -r include/* /usr/local/include/
 
 # VDMS install
-RUN cd / && wget https://github.com/intellabs/vdms/archive/v1.0.0.tar.gz && tar xf v1.0.0.tar.gz
-RUN mv vdms-1.0.0 vdms && cd vdms && mkdir db && scons -j16 INTEL_PATH=/
+RUN git clone https://github.com/intellabs/vdms/ && \
+    cd vdms && git checkout develop && \
```

Comment:
Will you be changing this to 2.0 checkout? Should this commit go after the merge?